File: scripts/timeTracking.py
import psutil
import time

# List of applications to monitor
applications_to_monitor = ['discord.exe', 'zoom.exe','speedtest.exe','sublime_text.exe', 'chrome.exe']

# Dictionary to store application runtime
application_runtime = {app: 0 for app in applications_to_monitor}

# Function to check if an application is running
def is_app_running(app_name):
    for process in psutil.process_iter(['pid', 'name']):
        if app_name.lower() in process.info['name'].lower():
            return True
    return False

# Main monitoring loop
try:
    while True:
        for app in applications_to_monitor:
            if is_app_running(app):
                application_runtime[app] += 1  # Increment runtime by 1 second
        time.sleep(1)  # Wait for 1 second before checking again
except KeyboardInterrupt:
    # Print the runtime when the script is stopped
    for app, runtime in application_runtime.items():
        print(f"{app} was running for {runtime} seconds.")

File: scripts/timeTracking.mjs
import psList from 'ps-list';
const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));

console.log('Monitoring started...');

// List of applications to monitor
const applicationsToMonitor = ['DW4Windows.exe', 'zoom', 'speedtest', 'sublime_text', 'chrome'];

// Dictionary to store application runtime
const applicationRuntime = applicationsToMonitor.reduce((acc, app) => {
  acc[app] = 0;
  return acc;
}, {});

// Function to check if an application is running
async function isAppRunning(appName) {
    const processes = await psList();
    return processes.some(process => process.name.toLowerCase().includes(appName.toLowerCase()));
}

// Main monitoring loop
(async () => {
    try {
        while (true) {
            for (const app of applicationsToMonitor) {
                if (await isAppRunning(app)) {
                    applicationRuntime[app] += 1; // Increment runtime by 1 second
                }
            }
            await sleep(1000); // Wait for 1 second before checking again
        }
    } catch (error) {
        console.error(error);
    } finally {
        // Print the runtime when the script is stopped
        for (const [app, runtime] of Object.entries(applicationRuntime)) {
            console.log(`${app} was running for ${runtime} seconds.`);
        }
    }
})();
